libs/stokesian_dynamics/multi_array.hpp

```diff
@@ -0,0 +1,248 @@
+#pragma once
+
+#include <cmath>
+#include <stdexcept>
+#include <string>
+#include <type_traits>
+
+#ifdef __CUDACC__
+#define DEVICE_FUNC __host__ __device__
+#else
+#define DEVICE_FUNC
+#endif
+
+#if __cplusplus == 201402L
+#define CXX14_CONSTEXPR constexpr
+#else
+#define CXX14_CONSTEXPR
+#endif
```

Comment:
C++14 is required anyway.

Reply:
So what should this be?

Reply:
I guess this is only needed if the library was to be used in a non-C++14 project. For Espresso use, the first branch is always taken. It should have a `>=` instead of an `==` though.

Reply:
just remove it and directly use ```constexpr``` where ```CXX14_CONSTEXPR``` occurred. You don't need it because we require c++14.

Reply:
We probably require C++14 throughout the code already, so just put `constexpr` directly  everywhere.